src/Repository/WorklogRepository.php

```diff
@@ -133,4 +133,23 @@ public function findBillableWorklogsByWorkerAndDateRange(string $workerIdentifie
             ])
             ->getQuery()->getResult();
     }
+
+    /**
+     * @throws \Exception
+     */
+    public function getWorklogsAttachedToInvoiceInDateRange(\DateTimeInterface $periodStart, \DateTimeInterface $periodEnd)
+    {
+        $from = new \DateTimeImmutable($periodStart->format('Y-m-d').' 00:00:00');
+        $to = new \DateTimeImmutable($periodEnd->format('Y-m-d').' 23:59:59');
+
+        return $this->createQueryBuilder('worklog')
```

Comment:
What happens if you have 1000 invoices and 1000000 worklogs?
This should be paginated and processed in pages.
Otherwise, there will be memory issues.

Reply:
See https://www.doctrine-project.org/projects/doctrine-orm/en/3.3/tutorials/pagination.html

Reply:
Probably something bad. I implemented this.